File: web/src/lib/types/plugins.ts
export interface Plugin {
	id: string;
	name: string;
	displayName: string;
	author: string;
	type: string;
	import: string;
	summary: string;
	iconUrl: string;
	bannerUrl: string;
	readme: string;
	latestVersion: string;
	versions: string[];
	stars: number;
	snippet: Record<string, string>;
	createdAt: string;
}

File: internal/traefik/plugins.go
package traefik

type Plugin struct {
	ID            string        `json:"id"`
	Name          string        `json:"name"`
	DisplayName   string        `json:"displayName"`
	Author        string        `json:"author"`
	Type          string        `json:"type"`
	Import        string        `json:"import"`
	Summary       string        `json:"summary"`
	IconURL       string        `json:"iconUrl"`
	BannerURL     string        `json:"bannerUrl"`
	Readme        string        `json:"readme"`
	LatestVersion string        `json:"latestVersion"`
	Versions      []string      `json:"versions"`
	Stars         int64         `json:"stars"`
	Snippet       PluginSnippet `json:"snippet"`
	CreatedAt     string        `json:"createdAt"`
}

type PluginSnippet struct {
	K8S  string `json:"k8s"`
	Yaml string `json:"yaml"`
	Toml string `json:"toml"`
}
